### DISCUSSION TITLE
Question about update schedule of dependencies

### DISCUSSION THREAD

**Discussion by PatrickZimmerer** [stephanrauh/ngx-extended-pdf-viewer#1976] [General]
Hey there, I created an Issue in the PDF.js repo and it got closed, now I only need to know whats the typical update schedule for this project to update the pdf.js versions, you can find the bug here https://github.com/mozilla/pdf.js/issues/17327 which is solved in here: https://github.com/mozilla/pdf.js/pull/17330

**Comment by stephanrauh** [comment]
Congratulations! And your bug fix even made it into the current release!

As for the schedule: well, at the moment I'm lagging behind. I hope to catch up before Christmas. However, version 4.0 is a breaking change, so I don't have a clue what to expect. It might be a rough ride.

Update: I've got sort of good news for you. I've managed to update to pdf.js 4.0 quickly. However, updating to the second 4.0 version (4.0.269, which has been released yesterday) seems to be a bummer. I've run into a dozen merge conflicts. I hope I can solve these conflicts easily - but not today, so I won't promise anything.

**Reply by stephanrauh** [reply]
@PatrickZimmerer  I've published version 18.2.0-alpha.0. The bleeding edge branch of it contains your bug fix. Updating to the new version wasn't easy, so I'm sure the new version is still buggy - that's why I made it an alpha version. Nonetheless, you can already test if your bug fix works.

**Comment by Musthafafazil** [comment]
Hello there,

I encountered an issue after upgrading from Angular 12 to Angular 16 and also updating ngx from version 16.2.8 to version 19. The problem manifests on iPad and Android tabs where the form fields aren't visible, although they appear fine on the web. Additionally, in the latest version, I noticed a peculiar issue with PDF form fields. Specifically, when I select "yes" on a radio button and save the form, upon reopening it, the radio button value isn't retained, unlike checkboxes and text boxes, whose values persist.

I would appreciate any insights or ideas on resolving these issues.

**Reply by stephanrauh** [reply]
Your radio button issue - is it the same as #2204?

**Reply by stephanrauh** [reply]
I've just tested the form demo with my iPad. The form fields show without problems. I ran it in Safari, but that's the only browser iOS supports, so I support even if you're implementing a Cordova application, the forms should render. What are you doing differently?

**Comment by Musthafafazil** [comment]
Hai Good Morning

Thank you for your email. I am currently using ngx version 18.0.0, and my
iPad is running version 16.x. In this version, all the form fields function
properly, as they are displayed when the form loads. However, after
updating to iPad version 17.x, the form fields are not initially visible
upon rendering. They only become visible when I zoom in or out. Notably,
I'm using the Chrome browser on iPad, and in version 16, the form fields
worked fine in Chrome.

Thanks and regards
Musthafa

**Reply by stephanrauh** [reply]
If you've got an Apple computer, you can connect your iPad with the cable and debug the CSS rules in Safari. That's a bit cumbersome, but it works well. Form fields are rendered as traditional HTML form fields, so you should be able to find out why the fields aren't visible quickly. My educated guess is that it's a CSS problem.
